### PR TITLE
Two-column rendering

### PR DESCRIPTION
**Pull request by jibbers42**
I've added the option for a more compact usage text that looks a bit more like many other outputs, like git, xsel, etc. The output is shown below.
It is enabled by using the following in you parser config:
`override def showCompactUsage = true`

Please let me know if I need to change anything to meet your approval.
Thanks!

```
scopt 3.x
Usage: scopt [update] [options] [<file>...]

  -f, --foo <value>         foo is an integer property
  -o, --out <file>          out is a required file property
  --max:<libname>=<max>     maximum count for <libname>
  -j, --jars <jar1>,<jar2>...
                            jars to include
  --kwargs k1=v1,k2=v2...   other arguments
  --verbose                 verbose is a flag
  --help                    prints this usage text
  <file>...                 optional unbounded args
some notes.

Command: update [options]
update is a command.
  -nk, --not-keepalive      disable keepalive
  --xyz <value>             xyz is a boolean property
```

### CODE REVIEW

**Comment by eed3si9n** [comment]
I like the general direction of this feature. I'm not sure if the name "compact" makes sense. "two columns" maybe? Also:

```
  --max:<libname>=<max>     maximum count for <libname>
  -j, --jars <jar1>,<jar2>...
                            jars to include
```

Does this imply that there's some cutoff for the column width that it decided to put `--jar`'s description on the next line? If so is the column width something the user can control?

**Comment by jibbers42** [comment]
I couldn't think of a great name so I went with compact as compared to the standard output (which I find a bit difficult to parse for some reason).

Yes, there is a cutoff of 25 chars + 2 spaces for the first column. It is dynamic so the description column will slide to be with two spaces of the max left column length (accounting for cutoff).

There isn't a way for the user to specify the max, but I guess that could be added to the config override or something similar.

Thanks for the quick response! Let me know if there is anything specific you want changed.

**Comment by jibbers42** [comment]
Oh, I forgot to ask, do you want it change to something like...
`override def showTwoColumnUsage = true`

That suggestion sounds fine to me.

**Comment by jibbers42** [comment]
Oops, I need to look at the tests. I think I looked at an sbt output for another project and thought the tests were passing before I pushed.

**Comment by eed3si9n** [comment]
> Oh, I forgot to ask, do you want it change to something like...
> 
> ```
> override def showTwoColumnUsage = true
> ```

How about something like

``` scala
def renderingMode: RenderingMode = RenderingMode.TwoColumns
```

where rendering mode is:

``` scala
trait RenderingMode // not sealed
object RenderingMode {
  case object OneColumn extends RenderingMode
  case object TwoColumns extends RenderingMode
}
```

It might be interesting if the methods are designed such that the user can add their own rendering mode.

**Comment by eed3si9n** [comment]
Also let's make two-column the default.

**Comment by jibbers42** [comment]
I'm not sure the best way to go about designing it to allow adding rendering modes. Any advice is appreciated - and feel free to dumb it down :)
I'll check back tomorrow sometime.

**Comment by eed3si9n** [comment]
In the overall code if `showTwoColumnUsage` is currently being used only within `def usage: String`, I was thinking something like this:

``` scala
def usage: String = renderUsage(renderingMode)
def renderUsage(mode: RenderingMode): String =
  mode match {
    case RenderingMode.OneColumn => renderOneColumnUsage
    case RenderingMode.TwoColumns => renderTwoColumnsUsage
  }
```

**Comment by jibbers42** [comment]
See if my latest commit is close to what you had in mind.

Your note to leave `RenderingMode` unsealed seemed to imply that a user could create a new `RenderingMode` and use it somehow. I'm not sure how to do that unless they override `renderUsage` and basically duplicate the case statements we have there, and add their own - is that the method?
I would think most people would override `usage` though. There is probably something I don't understand.

I also made a couple public defs so if somebody did try to render their own it would be at least a bit easier, see `optionsForRender` and `usageExample`

**Comment by eed3si9n** [comment]
> I would think most people would override usage though.

You're probably right, but I think the way the code looks makes the effects of the rendering mode clear.
In any case, thanks for the quick turn around of the updates. LGTM

**Comment by jibbers42** [comment]
Great, and thanks for the help. When do you plan to make a new release? (Obviously no emergency for this change, just curious)

**Comment by eed3si9n** [comment]
I guess sometime today after updating the README
